iroh-net/src/magicsock/node_map/endpoint.rs

```diff
@@ -171,6 +174,7 @@ impl Endpoint {
             direct_addr_state: BTreeMap::new(),
             last_used: options.active.then(Instant::now),
             last_call_me_maybe: None,
+            conn_type: Watchable::new(ConnectionType::None),
```

Comment:
my biggest concern with this approach is how "heavy" it might be to have a `Watchable` for each `node_map::Endpoint`

Reply:
It does acquire an additional `RwLock` write handle for each time you set it.  Same would be for a tokio `Watch`.  Though I think there would be very little contention for this, so it's probably fine?